portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -55,8 +61,10 @@ private void addEntity(String text, long timestamp)
             }
             else
             {
+                String email = (String) entity.getProperty("email");
```

Comment:
Better to flatten out logic:
 if (numComments == count)
          break;

After that you do not need else.